web/docs/database/operations/queries.md

```diff
@@ -0,0 +1,666 @@
+To prevent information leakage, the server won't forward these fields for any other HTTP status codes.
+
+### Using Entities in Queries
+
+In most cases, resources used in Queries will be [Entities](#entity).
+To use an Entity in your Query, add it to the `query` declaration in Wasp:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp {4,9} title="main.wasp"
+
+query getAllTasks {
+  fn: import { getAllTasks } from "@server/queries.js",
+  entities: [Task]
+}
+
+query getFilteredTasks {
+  fn: import { getFilteredTasks } from "@server/queries.js",
+  entities: [Task]
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp {4,9} title="main.wasp"
+
+query getAllTasks {
+  fn: import { getAllTasks } from "@server/queries.js",
+  entities: [Task]
+}
+
+query getFilteredTasks {
+  fn: import { getFilteredTasks } from "@server/queries.js",
+  entities: [Task]
+}
+```
+
+</TabItem>
+</Tabs>
+
+Wasp will inject the specified Entity into the Query's `context` argument, giving you access to the Entity's Prisma API:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```js title="src/server/queries.js"
+export const getAllTasks = async (args, context) => {
+  return context.entities.Task.findMany({});
+};
+
+export const getFilteredTasks = async (args, context) => {
+  return context.entities.Task.findMany({
+    where: { isDone: args.isDone },
+  });
+};
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="src/server/queries.ts"
+import { Task } from "@wasp/entities";
+import { GetAllTasks, GetFilteredTasks } from "@wasp/queries/types";
+
+export const getAllTasks: GetAllTasks<void, Task[]> = async (args, context) => {
+  return context.entities.Task.findMany({});
+};
+
+export const getFilteredTasks: GetFilteredTasks<
+  Pick<Task, "isDone">,
+  Task[]
+> = async (args, context) => {
+  return context.entities.Task.findMany({
+    where: { isDone: args.isDone },
+  });
+};
+```
+
+Again, annotating the Queries is optional, but greatly improves **full-stack type safety**.
+
+</TabItem>
+</Tabs>
+
+The object `context.entities.Task` exposes `prisma.task` from [Prisma's CRUD API](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/crud).
+
+## Options Reference
+
+### Declaring Queries
+
+The `query` declaration supports the following fields:
+
+- `fn: ServerImport` <Required />
+
+  The import statement of the Query's NodeJs implementation.
+
+- `entities: [Entity]`
+
+  A list of entities you wish to use inside your Query.
+  For instructions on using Entities in Queries, take a look at [the guide](#using-entities-in-queries).
+
+#### Example
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+Declaring the Query:
+
+```wasp
+query getFoo {
+    fn: import { getFoo } from "@server/queries.js"
+    entities: [Foo]
+}
+```
+
+Makes the following import on both the server and the client:
```

Comment:
Reads weird, "Makes the following import", does it import it somewhere or what?

We mention that it is available on the server, but don't really give them an example of importing a query on the server (I don't think we should give it) so maybe no mention server here?